source/backend/dal/Helpers/Extensions/AcquisitionExtensions.cs

```diff
@@ -76,8 +76,11 @@ private static IQueryable<Entities.PimsAcquisitionFile> GenerateCommonAcquisitio
                 query = query.Where(r => EF.Functions.Like(r.MinistryProjectName, $"%{filter.ProjectNameOrNumber}%") || EF.Functions.Like(r.MinistryProjectNumber, $"%{filter.ProjectNameOrNumber}%"));
             }
 
-            // Business Requirement: limit search results to user's assigned region(s)
-            query = query.Where(r => regions.Contains(r.RegionCode));
+            if (regions.Count > 0)
+            {
+                // Business Requirement: limit search results to user's assigned region(s)
+                query = query.Where(r => regions.Contains(r.RegionCode));
+            }
```

Comment:
I'm okay with this change but this would cause a change in the Acquisition List view where users with **no region associated** would see ALL acquisition files (instead of none). Are we sure we want to make this change @devinleighsmith ? 

I'm also a bit lost about this particular change because this method is only called from the Acquisition List View but the feature being implemented has not relation to that - it is about property information tab. Could you clarify why we need this change here @Maheshbabu28 ?
Thanks

Reply:
I was trying to get to PIMS  File tab while clicking on Acquisition from list view where nothing was loaded  which led me to this piece of code.

The query created when there is no **if**  condition
SELECT [p].[ACQUISITION_FILE_ID], [p].[ACQ_PHYS_FILE_STATUS_TYPE_CODE], [p].[ACQUISITION_FILE_STATUS_TYPE_CODE], [p].[ACQUISITION_FUNDING_TYPE_CODE], 
[p].[ACQUISITION_TYPE_CODE], [p].[APP_CREATE_TIMESTAMP], [p].[APP_CREATE_USER_DIRECTORY], [p].[APP_CREATE_USER_GUID], [p].[APP_CREATE_USERID], 
[p].[APP_LAST_UPDATE_TIMESTAMP], [p].[APP_LAST_UPDATE_USER_DIRECTORY], [p].[APP_LAST_UPDATE_USER_GUID], [p].[APP_LAST_UPDATE_USERID], [p].[ASSIGNED_DATE],
 [p].[CONCURRENCY_CONTROL_NUMBER], [p].[CPS_PRODUCT_CODE], [p].[DB_CREATE_TIMESTAMP], [p].[DB_CREATE_USERID], [p].[DB_LAST_UPDATE_TIMESTAMP], 
 [p].[DB_LAST_UPDATE_USERID], [p].[DELIVERY_DATE], [p].[FILE_NAME], [p].[FILE_NO], [p].[FILE_NUMBER], [p].[FUNDING_OTHER], [p].[MINISTRY_PROJECT_NAME],
 [p].[MINISTRY_PROJECT_NUMBER], [p].[REGION_CODE]
FROM [PIMS_ACQUISITION_FILE] AS [p]
WHERE 0 = 1

when we implement **if** condition 

SELECT [p].[ACQUISITION_FILE_ID], [p].[ACQ_PHYS_FILE_STATUS_TYPE_CODE], [p].[ACQUISITION_FILE_STATUS_TYPE_CODE],
 [p].[ACQUISITION_FUNDING_TYPE_CODE], [p].[ACQUISITION_TYPE_CODE], [p].[APP_CREATE_TIMESTAMP], [p].[APP_CREATE_USER_DIRECTORY],
 [p].[APP_CREATE_USER_GUID], [p].[APP_CREATE_USERID], [p].[APP_LAST_UPDATE_TIMESTAMP], [p].[APP_LAST_UPDATE_USER_DIRECTORY], 
 [p].[APP_LAST_UPDATE_USER_GUID], [p].[APP_LAST_UPDATE_USERID], [p].[ASSIGNED_DATE], [p].[CONCURRENCY_CONTROL_NUMBER], [p].[CPS_PRODUCT_CODE], 
 [p].[DB_CREATE_TIMESTAMP], [p].[DB_CREATE_USERID], [p].[DB_LAST_UPDATE_TIMESTAMP], [p].[DB_LAST_UPDATE_USERID], [p].[DELIVERY_DATE], [p].[FILE_NAME],
 [p].[FILE_NO], [p].[FILE_NUMBER], [p].[FUNDING_OTHER], [p].[MINISTRY_PROJECT_NAME], [p].[MINISTRY_PROJECT_NUMBER], [p].[REGION_CODE]
FROM [PIMS_ACQUISITION_FILE] AS [p]
WHERE [p].[ACQUISITION_FILE_STATUS_TYPE_CODE] = @__filter_AcquisitionFileStatusTypeCode_0

If no region exist for a user shall we show all or none? If none I can revert this code

Reply:
^^ question for @devinleighsmith or BAs, maybe?

Reply:
The user must belong to the region of an acquisition file in order to see that acquisition file. The current logic seen on the acquisition list view is correct. This is called out pretty clearly in https://jira.th.gov.bc.ca/browse/PSP-3252

This logic varies within psp-4297 however, on "PIMS Files" screen the user region limitation should be removed, and all acquisition files related to that property should be shown.

Reply:
@devinleighsmith done